packages/framework/aqueduct/src/data-objects/types.ts

```diff
@@ -40,3 +43,32 @@ export interface IDataObjectProps<I extends DataObjectTypes = DataObjectTypes> {
 	readonly providers: AsyncFluidObjectProvider<I["OptionalProviders"]>;
 	readonly initProps?: I["InitialState"];
 }
+
+/**
+ * A class that has a factory that can create a `DataObject` and a
```

Comment:
These docs should probably link createDataObjectKind . Also mention of "class" and "constructor" here is out of date.